core/trino-main/src/main/java/io/trino/sql/planner/optimizations/StreamPropertyDerivations.java

```diff
@@ -161,12 +162,9 @@ public static StreamProperties deriveProperties(
                 types,
                 typeAnalyzer);
 
-        StreamProperties result = node.accept(new Visitor(plannerContext.getMetadata(), session), inputProperties)
+        StreamProperties result = deriveStreamPropertiesWithoutOtherActualProperties(node, inputProperties, plannerContext.getMetadata(), session)
                 .withOtherActualProperties(otherProperties);
 
-        result.getPartitioningColumns().ifPresent(columns ->
-                verify(node.getOutputSymbols().containsAll(columns), "Stream-level partitioning properties contain columns not present in node's output"));
-
         Set<Symbol> localPropertyColumns = result.getLocalProperties().stream()
```

Comment:
this probably should be collected to `ImmutableSet`

Reply:
This set doesn't escape the local method context, so no particular need to make it immutable here (at least not as part of this PR).